lib/client/api_service.es6.js

```diff
@@ -11,90 +11,51 @@ require('../web_apis/release_interface_relationship.es6.js');
-  function getClientDAO(name, cls) {
-    return foam.dao.StreamingClientDAO.create({
-      of: cls,
-      delegate: foam.box.SubBox.create({
-        name: name,
-        delegate: foam.box.WebSocketBox.create({
-          // TODO(markdittmer): Explicitly select webSocketService port.
-          // (Current default: 4000.)
-          uri: 'ws://0.0.0.0:4000',
-        }, ctx),
-      }, ctx),
-    }, ctx);
-  }
-
-  const Release = org.chromium.apis.web.Release;
-  const WebInterface = org.chromium.apis.web.WebInterface;
-  const ReleaseWebInterfaceJunction =
-      org.chromium.apis.web.ReleaseWebInterfaceJunction;
-
-  let releaseDAO = foam.dao.EasyDAO.create({
-    name: C.RELEASE_NAME,
-    of: Release,
-    daoType: 'MDAO',
-  });
-  let webInterfaceDAO = foam.dao.EasyDAO.create({
-    name: C.WEB_INTERFACE_NAME,
-    of: WebInterface,
-    daoType: 'MDAO',
-  });
-  let releaseWebInterfaceJunctionDAO = getClientDAO(
-      C.RELEASE_WEB_INTERFACE_JUNCTION_NAME,
-      ReleaseWebInterfaceJunction);
-  let promises = [];
-  promises.push(getClientDAO(C.RELEASE_NAME, Release)
-      .select(foam.dao.AnonymousSink.create({
-        sink: {put: release => releaseDAO.put(release)},
-      })));
-  promises.push(getClientDAO(C.WEB_INTERFACE_NAME, WebInterface)
-      .select(foam.dao.AnonymousSink.create({
-        sink: {put: webInterface => webInterfaceDAO.put(webInterface)},
-      })));
-  let apiMatrix = org.chromium.apis.web.ApiMatrix.create({
-    releaseWebInterfaceJunctionDAO,
-    releaseDAO,
-    webInterfaceDAO,
-  },
-  // Provide a context that is aware to relationship DAOs.
-  // TODO(markdittmer): providing an interface for binding
-  // DAOs on Relationships.
-  foam.__context__.createSubContext({
-    releaseDAO,
-    webInterfaceDAO,
-    releaseWebInterfaceJunctionDAO,
-  }));
-
-  const ApiVelocityData = org.chromium.apis.web.ApiVelocityData;
-  const BrowserMetricData = org.chromium.apis.web.BrowserMetricData;
-  let apiVelocityDAO =
-      getClientDAO(C.API_VELOCITY_NAME, ApiVelocityData);
-  let failureToShipDAO =
-      getClientDAO(C.FAILURE_TO_SHIP_NAME, BrowserMetricData);
-  let browserSpecificDAO =
-      getClientDAO(C.BROWSER_SPECIFIC_NAME, BrowserMetricData);
-  let aggressiveRemovalDAO =
-      getClientDAO(C.AGGRESSIVE_REMOVAL_NAME, BrowserMetricData);
-  let apiConfluence = org.chromium.apis.web.ApiConfluence.create({
-    apiVelocityDAO,
-    failureToShipDAO,
-    browserSpecificDAO,
-    aggressiveRemovalDAO,
+  const ctx = foam.box.Context.create({
+    myname: '/page',
   });
+  const registry = ctx.registry;
+  const workerEvents = foam.core.FObject.create();
+  const workerEventsBox = foam.box.EventDispatchBox.create({
+    target: workerEvents,
+  }, ctx);
+  registry.register('events', null, workerEventsBox);
+
+  const workerMessagePortBox = foam.box.MessagePortBox.create({
+      target: new Worker('worker.bundle.js'),
+  }, ctx);
+  workerMessagePortBox.delegate;
```

Comment:
What is this doing?

Reply:
I should add a note/TODO here, and perhaps open a FOAM2 issue. This tickles a lazy factory that starts the page/worker `MessagePort` handshake (even though there are no data to send over it yet). That way the handshake cost is not incurred when we are trying to deliver data for the first time.

Reply:
TODO added.